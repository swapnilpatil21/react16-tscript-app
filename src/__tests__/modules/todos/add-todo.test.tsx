import React from 'react';
import { shallow } from "enzyme";
import AddTodo, { IAddTodo } from "../../../modules/todos/components/add-todo";

describe("<AddTodo />", () => {
  const defaultProps: IAddTodo = {
    addToDoHandler() {
      return;
    }
  };

  const addToDoHandler = jest.fn();

  it("should render correctly", () => {
    const wrapperComponent = shallow(<AddTodo {...defaultProps} />);

    expect(wrapperComponent).toMatchSnapshot();
  });

  it("doest not called add-todo fn if todo text empty", () => {
    const wrapperComponent = shallow(<AddTodo {...defaultProps} addToDoHandler={addToDoHandler} />);
    const fakeEvent = { preventDefault: () => {} };

    wrapperComponent.find('#add-todo-form').simulate('submit', fakeEvent)
    expect(addToDoHandler).toHaveBeenCalledTimes(0)
  })

  it("called add todo fn if todo text not empty", () => {
    const wrapperComponent = shallow(<AddTodo {...defaultProps} addToDoHandler={addToDoHandler} />);
    // created fakeEvent with a fake preventDefault function and pass it along
    const fakeEvent = { preventDefault: () => {} };

    wrapperComponent.find('input').simulate('change', {target: {value: 'test value'}});
    wrapperComponent.find('#add-todo-form').simulate('submit', fakeEvent)
    expect(addToDoHandler).toHaveBeenCalledTimes(1)
  })
});
