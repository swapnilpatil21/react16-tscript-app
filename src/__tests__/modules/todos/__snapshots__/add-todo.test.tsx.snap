// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodo /> should render correctly 1`] = `
<div>
  <p>
    <b>
      Add new Todo
    </b>
  </p>
  <form
    id="add-todo-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="new-todo"
    >
      New todo
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      id="add-todo-submit"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
